### ISSUE TITLE
feat: default value for source/(table with conn) 

### ISSUE THREAD

**Issue by st1page** [risingwavelabs/risingwave#17374]
Currently, the definition on the source/table with conn will not take effect

```SQL
create table t (
  x int,
  y int default 1,
  z timestamptz default now(),
) with (
  connector = 'kafka',
  topic = 't',
  properties.bootstrap.server = 'localhost:29092',
  scan.startup.mode = 'earliest'
) FORMAT PLAIN ENCODE JSON;
```

when insert `{"x":"1"}` to the topic, it will be resolved as 
```
dev=> select * from mv;
 x | y | z 
---+---+---
 1 |   | 
(1 row)
```

using `create source` will get the same result.

**Comment by BugenZhao** [comment]
If we're going to support it, we should also update this to pad the specified default values for parsing failures.

https://github.com/risingwavelabs/risingwave/blob/14b2f9c76ad20b7c644dfbf118f35205b9fe089d/src/connector/src/parser/mod.rs#L359

**Comment by xiangjinwu** [comment]
To clarify, does it only apply to json / csv where columns are listed in SQL?

For protobuf and avro, columns are derived and each encoding has their own specification about default values: [protobuf](https://protobuf.dev/programming-guides/proto3/#default) [avro](https://avro.apache.org/docs/1.11.1/specification/#:~:text=as%20defined%20above-,default,-%3A%20A%20default%20value)

**Comment by st1page** [comment]
> To clarify, does it only apply to json / csv where columns are listed in SQL?
> 
> For protobuf and avro, columns are derived and each encoding has their own specification about default values: [protobuf](https://protobuf.dev/programming-guides/proto3/#default) [avro](https://avro.apache.org/docs/1.11.1/specification/#:~:text=as%20defined%20above-,default,-%3A%20A%20default%20value)

Good point. I think for protobuf or avro, we need to give user an error to tell them the default value declared in the create statement does not take effect.
